Scanner.vue

```diff
@@ -90,42 +90,51 @@ export default {
   },
   methods: {
     async handleFileUpload(e) {
-      const url = URL.createObjectURL(e.target.files[0]);
-      this.image = new Image();
-      this.$q.loading.show({
+      const files = e.target.files;
+
+      // no file selected
+      if(files.length === 0) {
+        return;
+      }
+
+      const reader = new FileReader();
+      const vue = this;
+      vue.$q.loading.show({
         delay: 0, // ms
         spinner: Spinner,
         message: 'Scanning your photo! Hang on...'
       });
-      this.image.onload = async () => {
+      reader.onload = async function(e) {
+        const url = e.target.result;
+        vue.imageUrl = url;
```

Comment:
Why the switch away from an arrow function? It means (as you've done here) we have to keep track of `this` with a separate variable (which, btw, we usually use `self` for that purpose) which is confusing to developers.